protocols/kad/src/behaviour.rs

```diff
@@ -742,7 +762,13 @@ where
             step: ProgressStep::first(),
         };
         let peer_keys: Vec<kbucket::Key<PeerId>> = self.kbuckets.closest_keys(&target).collect();
-        self.queries.add_iter_closest(target, peer_keys, info)
+
+        if let Some(num_results) = num_results {
+            self.queries
+                .add_iter_closest_num_results(target, peer_keys, info, num_results)
```

Comment:
Alternatively, `num_result` can be included in `QueryInfo::GetClosestPeers` so the signatures of `continue_iter_closest` and `add_iter_closest` wouldn't change.